includes/data-port/models/class-sensei-import-course-model.php

```diff
@@ -279,4 +285,142 @@ private function delete_course_terms( $course_id, $taxonomy ) {
 			delete_post_meta( $course_id, '_module_order' );
 		}
 	}
+
+	/**
+	 * Migrates the imported post content to use the ids of the newly created lessons and modules.
+	 *
+	 * @param string $post_content The post content.
+	 *
+	 * @return string The migrated post content.
+	 */
+	private function migrate_post_contnet( $post_content ) {
+		if ( ! has_block( 'sensei-lms/course-outline', $post_content ) ) {
+			return $post_content;
+		}
+
+		$blocks = parse_blocks( $post_content );
+
+		$i = 0;
+		foreach ( $blocks as $block ) {
+			if ( 'sensei-lms/course-outline' === $block['blockName'] ) {
+				$mapped_block = $this->map_outline_block_ids( $block );
+				break;
+			}
+			$i++;
+		}
+		$blocks[ $i ] = $mapped_block;
+
+		return serialize_blocks( $blocks );
+	}
+
+	/**
+	 * Maps the ids of an outlined block to use the newly created values.
+	 *
+	 * @param array $outline_block The outline block.
+	 *
+	 * @return array The mapped block.
+	 */
+	private function map_outline_block_ids( $outline_block ) {
+		if ( empty( $outline_block['innerBlocks'] ) ) {
+			return $outline_block;
+		}
+
+		$mapped_inner_blocks = [];
+		foreach ( $outline_block['innerBlocks'] as $inner_block ) {
+			if ( 'sensei-lms/course-outline-module' === $inner_block['blockName'] ) {
+				$mapped_block = $this->map_module_block_id( $inner_block );
+			} elseif ( 'sensei-lms/course-outline-lesson' === $inner_block['blockName'] ) {
+				$mapped_block = $this->map_lesson_block_id( $inner_block );
+			} else {
+				$mapped_block = $inner_block;
+			}
+
+			if ( false !== $mapped_block ) {
+				$mapped_inner_blocks[] = $mapped_block;
+			}
+		}
+
+		$outline_block['innerBlocks'] = $mapped_inner_blocks;
+
+		return $outline_block;
+	}
+
+	/**
+	 * Map the ids of a lesson block.
+	 *
+	 * @param array $lesson_block The lesson block.
+	 *
+	 * @return bool|array The lesson block or false if the id couldn't be mapped.
+	 */
+	private function map_lesson_block_id( $lesson_block ) {
+		if ( empty( $lesson_block['attrs']['id'] ) ) {
+			return false;
+		}
+
+		// We first check for the lesson id to be a lesson which was imported during the import process. If that fails
+		// we check if the lesson already exists in the database. This could happen in case of a course update.
+		$lesson_id = $this->task->get_job()->translate_import_id( Sensei_Data_Port_Lesson_Schema::POST_TYPE, 'id:' . $lesson_block['attrs']['id'] );
+		if ( null === $lesson_id && null === $this->task->get_job()->translate_import_id( Sensei_Data_Port_Lesson_Schema::POST_TYPE, $lesson_block['attrs']['id'] ) ) {
```

Comment:
Shouldn't the fallback `translate_import_id` call be also assigned to $lesson_id?
Also in that case we could compare the post title with the title attribute to make sure it's the same one.

Reply:
> Shouldn't the fallback translate_import_id call be also assigned to $lesson_id?

Nope. In that case the id is the actual post id.

>  Also in that case we could compare the post title with the title attribute to make sure it's the same one.

Yeah I though about this too but having a lesson with an exact same post id by chance is very difficult. But probably an extra check wouldn't hurt.

Reply:
After double-checking indeed this code was wrong. I'll fix it in the newest version.